R/draw-dm.R

```diff
@@ -334,3 +325,76 @@ dm_get_colors <- function(dm) {
 dm_get_available_colors <- function() {
   c("default", colors())
 }
+
+
+#' Add info about a dm's tables
+#'
+#' @inheritParams dm_draw
+#' @param ... Descriptions for tables to set in the form `description = table`.
+#' These descriptions will be displayed when creating a diagram with [dm_draw()].
+#' Multi-line descriptions can be achieved using the newline symbol `\n`.
+#' The currently set descriptions can be checked using [dm_get_table_description()].
+#' Descriptions can be removed using `NULL = table`.
+#'
+#' `tidyselect` is supported, see [dplyr::select()] for details on the semantics.
+#'
+#' @return A `dm` object containing descriptions for specified tables.
+#' @export
+#'
+#' @examples
+dm_set_table_description <- function(dm, ...) {
+  check_not_zoomed(dm)
+  selected <- eval_select_table(quo(c(...)), src_tbls_impl(dm), unique = FALSE)
+  def <- dm_get_def(dm, quiet = TRUE)
+  reduce2(
+    selected,
+    names(selected),
+    function(def, table, desc) {
+      # `eval_select_table()` interprets the name `NULL` as "NULL"
+      # downside is, that no table description can be defined as "NULL"
+      desc <- if (desc == "NULL") {
+        NULL
+      } else {
+        desc
+      }
+      def$data[[which(def$table == table)]] <- structure(
+        def$data[[which(def$table == table)]],
+        "description" = desc
+      )
+      def
+    },
+    .init = def
+  ) %>%
+    new_dm3()
+}
+
+
+#' @inheritParams dm_add_pk
+#' @rdname dm_set_table_description
+#'
+#' @return A named vector of tables, with the descriptions in the names.
+#'
+#' @export
+#'
+#' @examples
+dm_get_table_description <- function(dm, table = NULL, ...) {
+  check_dots_empty()
+  check_not_zoomed(dm)
+
+  table_expr <- enexpr(table) %||% src_tbls_impl(dm, quiet = TRUE)
+  table_names <- eval_select_table(table_expr, set_names(src_tbls_impl(dm, quiet = TRUE)))
+
+  dm_get_table_description_impl(dm, table_names)
+}
+
+dm_get_table_description_impl <- function(dm, table_names) {
+  def <- dm_get_def(dm, quiet = TRUE)
+  map(
+    table_names,
+    function(table_name) {
+      attr(def[def$table == table_name,]$data[[1]], "description")
```

Comment:
The labelled package uses `"label"` :

```suggestion
      attr(def$data[[which(def$table == table_name)]], "label")
```

Can you use `eval_select_both()` to work with table positions instead of table names?